server/selective_sync.go

```diff
@@ -6,25 +6,24 @@ import (
 	"github.com/mattermost/mattermost/server/public/model"
 )
 
-// ChatMembersSpanPlatforms determines if the members of the given channel span both Mattermost and
+// ChatShouldSync determines if the members of the given channel span both Mattermost and
 // MS Teams. Chats between users on the same platform are skipped if selective sync is enabled.
-func (p *Plugin) ChatSpansPlatforms(channelID string) (bool, *model.AppError) {
+// Chats with only a single member are self chats and always sync.
+func (p *Plugin) ChatShouldSync(channelID string) (bool, *model.AppError) {
 	members, appErr := p.API.GetChannelMembers(channelID, 0, math.MaxInt32)
 	if appErr != nil {
 		return false, appErr
 	}
+	if len(members) == 1 {
+		return true, nil
+	}
 
 	return p.ChatMembersSpanPlatforms(members)
 }
 
 // ChatMembersSpanPlatforms determines if the given channel members span both Mattermost and
 // MS Teams. Chats between users on the same platform are skipped if selective sync is enabled.
-// Chats with only a single member are self chats and always sync.
 func (p *Plugin) ChatMembersSpanPlatforms(members model.ChannelMembers) (bool, *model.AppError) {
-	if len(members) == 1 {
-		return true, nil
-	}
-
```

Comment:
I considered throwing an error here, if only one member is passed. As this function really shouldn't be called for a single user...based on name. Thoughts?

Reply:
I thunk it makes sense to return an error in that case

Reply:
Added error.